src/OpenColorIO/Platform.cpp

```diff
@@ -197,6 +197,42 @@ std::string CreateTempFilename(const std::string & filenameExt)
     return filename;
 }
 
+std::ifstream CreateIfstream(const char * filename, std::ios_base::openmode mode)
+{
+#ifdef _WIN32
+    return std::ifstream(Utf8ToUtf16(filename).c_str(), mode);
+#else
+    return std::ifstream(filename, mode);
+#endif
+}
+
+void OpenIfstream(std::ifstream * stream, const char * filename, std::ios_base::openmode mode)
+{
+#ifdef _WIN32
+    stream->open(Utf8ToUtf16(filename).c_str(), mode);
+#else
+    stream->open(filename, mode);
+#endif
+}
+
+#ifdef _WIN32
+std::wstring Utf8ToUtf16(std::string str)
+{
+    int sz = MultiByteToWideChar(CP_UTF8, 0, &str[0], (int)str.size(), NULL, 0);
```

Comment:
[Question] As the minimal C++ version for the project is now C++11, could the code use some new C++ feature? 

I found `std::codecvt_utf8_utf16` which provides a platform agnostic implementation. Unfortunately it's deprecated from C++17.

I also check OpenImageIO library which uses the proposed implementation with the same defines everywhere is the code.

Reply:
I used [godbolt](https://godbolt.org/) for investigation, and no recent major compilers flag the feature as deprecated!.